src/editor/scripting/builtin.cc

```diff
@@ -0,0 +1,2796 @@
+   {"random_1",
+    BuiltinFunctionInfo(
+       /** TRANSLATORS: max is the name of a function parameter */
+       []() { return _("Returns a random value between 1 and max."); },
+       new FunctionBase("math.random",
+                        VariableType(VariableTypeID::Nil),      // call on
+                        VariableType(VariableTypeID::Integer),  // returns
+                        {std::make_pair("max", VariableType(VariableTypeID::Integer))}))},
+   {"random_2",
+    BuiltinFunctionInfo(
+       /** TRANSLATORS: min and max are names of function parameters */
+       []() { return _("Returns a random value between min and max."); },
+       new FunctionBase("math.random",
+                        VariableType(VariableTypeID::Nil),      // call on
+                        VariableType(VariableTypeID::Integer),  // returns
+                        {std::make_pair("min", VariableType(VariableTypeID::Integer)),
+                         std::make_pair("max", VariableType(VariableTypeID::Integer))}))},
+
+   // Shipped Lua functions
+
+   {"sleep",
+    BuiltinFunctionInfo(
+       []() { return _("Pauses the current thread for the given number of milliseconds."); },
+       new FunctionBase("sleep",
+                        VariableType(VariableTypeID::Nil),  // call on
+                        VariableType(VariableTypeID::Nil),  // returns
+                        {std::make_pair("milliseconds", VariableType(VariableTypeID::Integer))}),
+       "scripting/coroutine.lua")},
+   {"wake_me",
+    BuiltinFunctionInfo(
+       []() {
+	       return _(
+	          "Pauses the current thread and resumes it at the given gametime (in milliseconds).");
+       },
+       new FunctionBase("wake_me",
+                        VariableType(VariableTypeID::Nil),  // call on
+                        VariableType(VariableTypeID::Nil),  // returns
+                        {std::make_pair("time", VariableType(VariableTypeID::Integer))}),
+       "scripting/coroutine.lua")},
+
+   // TODO(Nordfriese): This function can take varargs
+   {"array_combine",
+    BuiltinFunctionInfo(
+       []() { return _("Concatenates the given arrays into a single array."); },
+       new FunctionBase("array_combine",
+                        VariableType(VariableTypeID::Nil),  // call on
+                        VariableType(VariableType(VariableTypeID::Integer),
+                                     VariableType(VariableTypeID::Any)),  // returns
+                        {std::make_pair("array1",
+                                        VariableType(VariableType(VariableTypeID::Integer),
+                                                     VariableType(VariableTypeID::Any))),
+                         std::make_pair("array2",
+                                        VariableType(VariableType(VariableTypeID::Integer),
+                                                     VariableType(VariableTypeID::Any)))}),
+       "scripting/table.lua")},
+
+   {"connected_road",
```

Comment:
"connected_road" should be shifted from data/scripting to src/scripting. It will become unmaintainable if we reference functions in the C++ that don't exist in the C++ code.

Note though that the tribes-filtering branch has a bugfix for placing carriers on roads that will conflict with this refactoring.